Shell/LineEditor.cpp

```diff
@@ -161,25 +172,39 @@ void LineEditor::tab_complete_other_token(String& token)
         path = g.cwd;
     }
 
+    // This is a bit naughty, but necessary without reordering the loop
+    // below. The loop terminates early, meaning that
+    // the suggestions list is incomplete.
+    // We only do this if the token is empty though.
+    if (token.is_empty()) {
+        CDirIterator suggested_files(path, CDirIterator::SkipDots);
+        while (suggested_files.has_next()) {
+            suggestions.append(suggested_files.next_path());
+        }
+
+        return suggestions;
```

Comment:
We can't quite do this. Empty tokens need to be processed by the next loop. This is how we get single matches, i.e. when a directory has one file.
Try this:
```
mkdir test
touch test/test
cat test/<tab>
```
This is supposed to complete to `test/test`.

Reply:
Ahh of course, I'll fix that one up.